tests/components/file_upload/__init__.py

```diff
@@ -1 +1,4 @@
 """Tests for the File Upload integration."""
+import pathlib
+
+TEST_LARGE_FILE = pathlib.Path(__file__).parent / "test_large_file.txt"
```

Comment:
Instead of including a large file in the repo, generate one on the fly in the test. You can use [`StringIO`](https://docs.python.org/3/library/io.html#io.StringIO)

Reply:
I think that you can even just implement BaseIO and generate the string on the fly.

Reply:
I used `tempfile` for it and made it a fixture. Same result without a checked in file.

Reply:
The problem is that this does a lot of I/O. At least uploading we should be able to use a virtual file.